plugins/destination/meilisearch/client/read.go

```diff
@@ -45,17 +57,127 @@ func (c *Client) Read(_ context.Context, table *schema.Table, sourceName string,
 	return nil
 }
 
-func unmap(table *schema.Table) func(hit any) ([]any, error) {
-	return func(hit any) ([]any, error) {
-		m, ok := hit.(map[string]any)
-		if !ok {
-			return nil, fmt.Errorf("unsupported format for doc: %T", hit)
-		}
+func unmap(sc *arrow.Schema) func(m map[string]any) (arrow.Record, error) {
```

Comment:
What is this function - `unmap` ? can we avoid using it and use `arrow.Field` directly without having a function that return a function?

Reply:
We read `map[string]any` (JSON) from Meilisearch and need to transform this to the comparable (record) format.

Reply:
Moved from closure to separate func in 8faf82a